components/SubmissionForm.vue

```diff
@@ -198,7 +220,48 @@ const validateFields = (e: Event) => {
         e.preventDefault()
         return
     }
-    submissionStore.submit()
+    submitNewSubmission()
```

Comment:
should await this function

Reply:
also, this shouln't be called in validateFields. It should be the other way around where submit should call validateFields.